### PR TITLE
add language support to translation client, solves #1763

### PR DESCRIPTION
**Pull request by ceferisbarov**

### CODE REVIEW

**Review by Wauplin — COMMENTED** [review]
Hi @ceferisbarov thanks for opening the PR! It is already in good shape with the logic + docs + tests :rocket: I've added a few comments below. Other than that there are 2 things to do:
1. There is a `InferenceClient` and an `AsyncInferenceClient`. When the logic is updated in one, the other should be updated as well. Luckily we have a script to automatically port the logic to the async client. You should first install `pip install -e ".[dev]"` locally (see [here](https://huggingface.co/docs/huggingface_hub/installation#editable-install)). Then run  `make style` which will apply the code linter + code formatter + the script to generate async client. Once done, commit the changes and that's it.
2. The tests in `test_inference_client.py` are mocked. This means that the requests are not really made to the Inference server each time you run the test. What happens is that the first time you run the test, it will create a file locally under `./tests/cassettes`. The next times you run the test, this file will be reused. What you need to do is to run the test to create the mock file automatically, then check the content "looks good" and finally commit it to the repo.

If you need any help for one of those things, just let me know :hugs:

**Comment by ceferisbarov** [comment]
@Wauplin I didn't know abot test cassettes. Cool stuff. Can you have another look at the PR? Thanks.

File: src/huggingface_hub/inference/_client.py [hunk 1]
```diff
@@ -1733,8 +1743,25 @@ def translation(self, text: str, *, model: Optional[str] = None) -> str:
         >>> client.translation("My name is Wolfgang and I live in Berlin", model="Helsinki-NLP/opus-mt-en-fr")
         "Je m'appelle Wolfgang et je vis à Berlin."
         ```
+
+        Specifying languages:
+        ```py
+        >>> client.translation("My name is Sarah Jessica Parker but you can call me Jessica", model="facebook/mbart-large-50-many-to-many-mmt", "src_lang="en_XX", tgt_lang="fr_XX")
```
Comment (Wauplin) [inline src/huggingface_hub/inference/_client.py:1749]: typo

```suggestion
        >>> client.translation("My name is Sarah Jessica Parker but you can call me Jessica", model="facebook/mbart-large-50-many-to-many-mmt", src_lang="en_XX", tgt_lang="fr_XX")
```

File: src/huggingface_hub/inference/_client.py [hunk 2]
```diff
@@ -1700,12 +1700,16 @@ def token_classification(self, text: str, *, model: Optional[str] = None) -> Lis
...
+        self, text: str, *, model: Optional[str] = None, src_lang: Optional[str] = None, tgt_lang: Optional[str] = None
+    ) -> str:
         """
         Convert text from one language to another.
 
         Check out https://huggingface.co/tasks/translation for more information on how to choose the best model for
-        your specific use case. Source and target languages usually depends on the model.
+        your specific use case. Source and target languages usually depend on the model.
+        However, it is possible to specify source and target languages for certain models. If you are working with one of these models,
+        you can use `src_lang` and `tgt_lang` arguments to pass the relevant information.
```
Comment (Wauplin) [inline src/huggingface_hub/inference/_client.py:1712]:
```suggestion
        you can use `src_lang` and `tgt_lang` arguments to pass the relevant information.
        You can find this information in the model card.
```

**Review by Wauplin — APPROVED** [review]
Looks good! Thanks for making the changes and uploading the VCR cassette :fire

File: src/huggingface_hub/inference/_generated/_async_client.py [hunk 3]
```diff
@@ -1725,12 +1725,16 @@ async def token_classification(self, text: str, *, model: Optional[str] = None)
...
+        self, text: str, *, model: Optional[str] = None, src_lang: Optional[str] = None, tgt_lang: Optional[str] = None
+    ) -> str:
         """
         Convert text from one language to another.
 
         Check out https://huggingface.co/tasks/translation for more information on how to choose the best model for
-        your specific use case. Source and target languages usually depends on the model.
+        your specific use case. Source and target languages usually depend on the model.
+        However, it is possible to specify source and target languages for certain models. If you are working with one of these models,
+        you can use `src_lang` and `tgt_lang` arguments to pass the relevant information.
```
Comment (Wauplin) [inline src/huggingface_hub/inference/_generated/_async_client.py:1737]:
```suggestion
        you can use `src_lang` and `tgt_lang` arguments to pass the relevant information.
        You can find this information in the model card.
```

File: src/huggingface_hub/inference/_generated/_async_client.py [hunk 4]
```diff
@@ -1759,8 +1769,25 @@ async def translation(self, text: str, *, model: Optional[str] = None) -> str:
         >>> await client.translation("My name is Wolfgang and I live in Berlin", model="Helsinki-NLP/opus-mt-en-fr")
         "Je m'appelle Wolfgang et je vis à Berlin."
         ```
+
+        Specifying languages:
+        ```py
+        >>> client.translation("My name is Sarah Jessica Parker but you can call me Jessica", model="facebook/mbart-large-50-many-to-many-mmt", "src_lang="en_XX", tgt_lang="fr_XX")
```
Comment (Wauplin) [inline src/huggingface_hub/inference/_generated/_async_client.py:1775]: typo
```suggestion
        >>> client.translation("My name is Sarah Jessica Parker but you can call me Jessica", model="facebook/mbart-large-50-many-to-many-mmt", src_lang="en_XX", tgt_lang="fr_XX")
```

**Review by Wauplin — APPROVED** [review]
I've made 2 small changes (typo + add comment about model card).
Let's wait for the CI to complete and it'll be good to merge :) Some tests are failing for it's not related to the PR.

**Pull request merged**